### PR TITLE
types(Options): add types for cacheWithLimits

### PR DESCRIPTION
**Pull request by DTrombett**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds typings for the `limits` parameter in the `cacheWIthLimits` method and improve the description of the `makeCache` property in `ClientOptions`, replacing the old one that was still there.
Tests also updated.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

File: typings/index.d.ts [hunk 1]
```diff
@@ -2726,8 +2726,32 @@ export type BitFieldResolvable<T extends string, N extends number | bigint> =
...
+  | 'ReactionManager'
+  | 'ReactionUserManager'
+  | 'RoleManager'
+  | 'StageInstanceManager'
+  | 'ThreadManager'
+  | 'ThreadMemberManager'
+  | 'UserManager'
+  | 'VoiceStateManager';
+
 export type CacheFactory = <T>(manager: { name: string }, holds: { name: string }) => Collection<Snowflake, T>;
```
Comment (1Computer1) [inline typings/index.d.ts:2749]: Might as well add types to this too.

**Review by iCrawl — APPROVED** [review]

File: typings/index.d.ts (hunk 1 above)
Comment (DTrombett) [reply typings/index.d.ts]: Done, I think it's a mess lol, but it works.

**Comment by DTrombett** [comment]
Should I wait for #6110 to be merged to add types for new methods too?

File: typings/index.d.ts [hunk 2]
```diff
@@ -2725,7 +2725,54 @@ export type BitFieldResolvable<T extends string, N extends number | bigint> =
 
 export type BufferResolvable = Buffer | string;
 
-export type CacheFactory = <T>(manager: { name: string }, holds: { name: string }) => Collection<Snowflake, T>;
+export type CachedManagerTypes =
```
Comment (SpaceEEC) [inline typings/index.d.ts:2728]: This could be `keyof CacheFactoryArgs`, but it does not seem to be used anywhere (anymore?)?

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

File: typings/index.d.ts (hunk 2 above)
Comment (DTrombett) [reply typings/index.d.ts:2728]: This what? The generic type `T`?

Comment (DTrombett) [reply typings/index.d.ts:2728]: Ohh, got it. You're right!

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**